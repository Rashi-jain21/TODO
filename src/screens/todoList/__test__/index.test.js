import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import TODOList from '..';
import {FlatList} from 'react-native';

const mockStore = configureStore();

it('renders with snapshot with no data', () => {
  const initialState = {todos: []};
  const tree = renderer
    .create(
      <Provider store={mockStore(initialState)}>
        <TODOList />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders todo list with data', () => {
  const initialState = {
    todos: [{name: 'item1', id: '1234', status: 'complete'}],
  };
  const tree = renderer.create(
    <Provider store={mockStore(initialState)}>
      <TODOList />
    </Provider>,
  );
  expect(tree.toJSON()).toMatchSnapshot();
  expect(tree.root.findByType(FlatList).children).toHaveLength(1);
});
