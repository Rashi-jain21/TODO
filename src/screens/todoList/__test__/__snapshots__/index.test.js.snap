// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "space-evenly",
      "paddingTop": 70,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#8c8eab",
        "height": 1,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 16,
          "fontWeight": "500",
        }
      }
    >
      No Data
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#4290f5",
        "borderRadius": 10,
        "height": 40,
        "justifyContent": "center",
        "position": "absolute",
        "right": 20,
        "top": 20,
        "width": 100,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "white",
              },
            ]
          }
        >
          Add
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders with snapshot with data 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "space-evenly",
      "paddingTop": 70,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#8c8eab",
        "height": 1,
      }
    }
  />
  <RCTScrollView
    data={
      Array [
        Object {
          "id": "1234",
          "name": "item1",
          "status": "complete",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "borderBottomWidth": 1,
              "borderColor": "#8c8eab",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "padding": 20,
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "green",
                    "fontSize": 20,
                  },
                  undefined,
                  Object {
                    "fontFamily": "anticon",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
          <Text
            style={
              Object {
                "alignSelf": "center",
                "fontSize": 16,
                "fontWeight": "500",
                "paddingLeft": 10,
              }
            }
          >
            item1
          </Text>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignSelf": "center",
                "position": "absolute",
                "right": 10,
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#900",
                    "fontSize": 20,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#4290f5",
        "borderRadius": 10,
        "height": 40,
        "justifyContent": "center",
        "position": "absolute",
        "right": 20,
        "top": 20,
        "width": 100,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "white",
              },
            ]
          }
        >
          Add
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders with snapshot with no data 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "space-evenly",
      "paddingTop": 70,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#8c8eab",
        "height": 1,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 16,
          "fontWeight": "500",
        }
      }
    >
      No Data
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#4290f5",
        "borderRadius": 10,
        "height": 40,
        "justifyContent": "center",
        "position": "absolute",
        "right": 20,
        "top": 20,
        "width": 100,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "white",
              },
            ]
          }
        >
          Add
        </Text>
      </View>
    </View>
  </View>
</View>
`;
